pkg/sql/sem/tree/stmt.go

```diff
@@ -101,6 +101,18 @@ func CanModifySchema(stmt Statement) bool {
 	return ok && scm.modifiesSchema()
 }
 
+// CanWriteData returns true if the statement can modify data.
+func CanWriteData(stmt Statement) bool {
+	if stmt.StatementType() == RowsAffected {
+		return true
+	}
+	switch stmt.(type) {
+	case *Insert, *Delete, *Update:
+		return true
+	}
+	return false
+}
+
```

Comment:
Can our `[...]` syntax evade this check? I think we might need to recurse? Also
should sequence functions like `nextval` etc. count as writes within this context?
cc @vilterp

Reply:
We call this in `newPlan` which does get recursively called for all statements, so I *think* we're set here.

Reply:
Ahh true, I missed that 👍

Reply:
yeah, `SELECT nextval('foo')` can cause a write, but those functions also spawn their own transaction, which immediately completes (you can't roll it back). So maybe the containing transaction would still sort of be read only?